libcaf_core/caf/serialization.test.cpp

```diff
@@ -861,6 +884,55 @@ OUTLINE("serializing a type that initializes members to a non-empty state") {
   )_";
 }
 
+// TODO: update tests to use apply instead of value overload
+SCENARIO("binary serializer and deserializer handle vectors of booleans") {
+  GIVEN("a binary serializer") {
+    auto sink = binary_serializer_wrapper{sys};
+    WHEN("serializing a vector of booleans with 8 values") {
+      auto val = std::vector<bool>{true,  false, true, true,
+                                   false, false, true, false};
+      check(sink.sink.value(val));
+      THEN("deserializing the result produces the value again") {
+        auto source = binary_deserializer{sys, sink.buffer};
+        std::vector<bool> copy = {};
```

Comment:
For keeping a consistent style:

```suggestion
        auto copy = std::vector<bool>{};
```

Same in the occurrences below.